docs/DeveloperGuide.md

```diff
@@ -416,6 +416,67 @@ The following sequence diagram shows how the add operation works:
 
 ![AddCommandSequenceDiagram-2](images/AddCommandSequenceDiagram-2.png)
 
+### Task Command
+
+#### Description
+
+The `task` command allows users to check and view all the tasks belonging to a particular student.
+During the execution of the `task` command, the user's input is being parsed in `AddressBookParser`.
+After which, a new `TaskCommand` object will be created, and is subsequently executed by the LogicManager.
+
+#### Implementation
+
+1. Upon receiving the user input,
+   the `LogicManager` starts to parse the given input text using `AdddressBookParser#parseCommand()`.
+2. The `AddressBookParser` invokes the respective `Parser` based on the first word of the input text.
+3. Since the first word in the user input matches the word "task", `TaskCommandParser#parse(arguments)` will be called.
+   In this case, the arguments refer to the remaining input text after the exclusion of the command word ("task").
+4. In the `TaskCommandParser#parse(arguments)`, the arguments will be tokenized into a `ArgumentMultimap`,
+   by using `ArgumentTokenizer#tokenize(String argsString, Prefex... prefixes)`.
+
+   <div markdown="span" class="alert alert-info">:information_source: 
+    **Note:** A ParseException will be thrown if the prefix of `StudentId` is missing, as it is a compulsory field.
+   </div>
+
+5. The `TaskCommandParser` will pass the studentId input (found in the `ArgumentMultimap`)
+   into `ParserUtil#parseStudentId(String studentId).`
+
+   <div markdown="span" class="alert alert-info">:information_source: 
+   **Note:** A NullException will be thrown if the supplied string argument is null.
+   </div>
+
+6. In `ParserUtil#parseStudentId(String studentId)`, the supplied argument will be trimmed using `String#trim()`.
+7. `StudentId#isValidId(String studentId)` will then be invoked,
+   which checks if the trimmed argument is valid (according to the Regex supplied).
+   If the argument is valid, a new StudentId object will be created and returned to the `TaskCommandParser`.
+   If the argument is not valid, a `ParseException` will be thrown.
+8. A new `TaskCommand` will be created (using the `StudentId` object created in Step 7) and returned to the `LogicManager`.
+9. The `LogicManager` will then call `TaskCommand#execute(Model model)`.
+10. In the `TaskCommand`, a `StudentIdContainsKeywordsPredicate` is created using the studentId.
+11. The `model#updateFilteredPersonList(predicate)` is then invoked such that the list is filtered by the predicate created.
+12. The `model#getFilteredPersonList()` is then called. 
+    The size of the resulting list determines if there is any `Person` that corresponds to the specified `studentId`. 
+    If the size is equals to 0, a `CommandException` will be thrown.
+
+    <div markdown="span" class="alert alert-info">:information_source: 
+    **Note:** The studentId is a UNIQUE field in TAPA.
```

Comment:
```suggestion
    **Note:** The studentId is a UNIQUE field in TAPA, so multiple students cannot have the same studentId.
```

Perhaps this would make this note clearer for readers?